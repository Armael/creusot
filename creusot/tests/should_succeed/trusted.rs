// SHOULD_SUCCEED: parse-print
#![feature(register_tool, rustc_attrs)]
#![register_tool(creusot)]
#![feature(proc_macro_hygiene, stmt_expr_attributes)]

extern crate creusot_contracts;

use creusot_contracts::*;

#[trusted]
fn omg() {}

#[trusted]
#[ensures(result == 10u32)]
fn omg2() -> u32 {
  5 // im evil
}

