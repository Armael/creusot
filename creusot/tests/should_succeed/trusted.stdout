module Type
  use Ref
  use mach.int.Int
  use mach.int.Int32
  use mach.int.Int64
  use mach.int.UInt32
  use mach.int.UInt64
  use string.Char
  use floating_point.Single
  use floating_point.Double
  use prelude.Prelude
end
module Trusted_Omg_Interface
  val omg () : ()
end
module Trusted_Omg
  val omg () : ()
end
module Trusted_Omg2_Interface
  use mach.int.Int
  use mach.int.UInt32
  val omg2 () : uint32
    ensures { result = (10 : uint32) }
    
end
module Trusted_Omg2
  use mach.int.Int
  use mach.int.UInt32
  val omg2 () : uint32
    ensures { result = (10 : uint32) }
    
end
